sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -289,7 +289,7 @@ static int subscribe()
        = MQTTClient_subscribe(mqtt_client, AZ_IOT_PROVISIONING_CLIENT_REGISTER_SUBSCRIBE_TOPIC, 1))
       != MQTTCLIENT_SUCCESS)
   {
-    printf("Failed to subscribe to the register topic filter, MQTTClient return code %d\n", rc);
+    (void)printf("Failed to subscribe to the register topic filter; MQTTClient return code %d .\n", rc);
```

Comment:
Why do we need all these `(void)` casts for every `printf` statement?

Reply:
Per Azure SDK C guidelines.  https://azure.github.io/azure-sdk/clang_implementation.html#clang-check-syscall-errors
![image](https://user-images.githubusercontent.com/64985898/86990753-27f3f580-c152-11ea-8836-71b3d4bbdb88.png)